ksql-serde/src/main/java/io/confluent/ksql/serde/delimited/KsqlDelimitedSerdeFactory.java

```diff
@@ -60,4 +74,23 @@ public KsqlDelimitedSerdeFactory() {
 
     return deserializer;
   }
+
+  private static class SchemaValidator implements SchemaWalker.Visitor<Void, Void> {
+
+    public Void visitPrimitive(final Schema schema) {
+      // Primitive types are allowed.
+      return null;
+    }
+
+    public Void visitBytes(final Schema schema) {
+      // Decimal type is allowed.
+      return null;
+    }
+
+    public Void visitSchema(final Schema schema) {
+      final String typeString = schema.type() == Type.BYTES ? "DECIMAL" : schema.type().toString();
+      throw new KsqlException("The '" + Format.DELIMITED
+          + "' format does not support type '" + typeString + "'");
```

Comment:
this is confusing - does it or does it not support "DECIMAL"? The comment above suggests that it does (and again, we should check with `DecimalUtil#isDecimal`)

Reply:
C&P error, fixed.